src/benchmarks/micro/libraries/System.Text.Encodings.Web/Perf.DefaultJavaScriptEncoder.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Text.Encodings.Web;
+
+namespace System.Text.Tests
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class Perf_DefaultJavaScriptEncoder
+    {
+        [Params("no escaping required", "&Hello+<World>!")]
+        public string sourceString;
+
+        // pads the string with a pseudorandom number of non-escapable characters
+        [Params(16, 512)]
+        public int paddingSize;
+
+        private JavaScriptEncoder _encoder;
+        private string _sourceBufferUtf16;
+        private char[] _destinationBufferUtf16;
+
+        private byte[] _sourceBufferUtf8;
+        private byte[] _destinationBufferUtf8;
+
+        [GlobalSetup]
+        public void SetupGetBytes()
+        {
+            _encoder = JavaScriptEncoder.Default;
+            _sourceBufferUtf16 = BuildSourceString();
+            _destinationBufferUtf16 = new char[paddingSize + 10 * sourceString.Length];
+
+            _sourceBufferUtf8 = Encoding.UTF8.GetBytes(_sourceBufferUtf16);
+            _destinationBufferUtf8 = new byte[paddingSize + 10 * sourceString.Length];
+            
+            string BuildSourceString()
```

Comment:
>BuildSourceString [](start = 19, length = 17)

can we have a common method somewhere that takes parameters from each of the encoders test?

Reply:
I resisted extracting shared components into a helper class, since it seems that benchmarks are pretty self-contained in this project. @adamsitnik might be able to comment about DRY concerns here.